plugin/src/main/java/io/deephaven/plugin/PluginAdapter.java

```diff
@@ -0,0 +1,25 @@
+package io.deephaven.plugin;
+
+import java.util.Objects;
+
+/**
+ * Adapts a {@link Plugin} into a {@link Registration}.
+ */
+public final class PluginAdapter implements Registration {
```

Comment:
Why does this class exist, rather than making Plugin implement Registration, offer it as a default method? Then `Plugin`s could be bound as `Registration`s instead.

As far as I can see, `Set<Plugin>` isn't injected except to adapt them - if `Plugin`s and `ObjectType`s were collected (from service loaders, other dagger bindings) straightaway as `Registration`s, I don't see anything we would lose within DH, and it isn't clear to me that outside of service loaders we anticipate users directly managing the set of possible plugins/objecttypes/registrations (except perhaps _as_ registrations)?

Reply:
Deleted.